orttraining/orttraining/training_api/checkpoint.cc

```diff
@@ -248,20 +257,32 @@ Status FromModuleState(const ModuleCheckpointState& module_state,
     return Status::OK();
   }
 
-  InlinedHashMap<std::string, OrtValue> requires_grad;
+  size_t num_requires_grad_params = 0;
+  size_t num_frozen_params = 0;
+  for ([[maybe_unused]] const auto& [name, value] : module_state.named_parameters) {
+    if (value->RequiresGrad()) {
+      num_requires_grad_params++;
```

Comment:
just FYI, [std::count_if](https://en.cppreference.com/w/cpp/algorithm/count) might be useful